common/utils/query_utils.ts

```diff
@@ -0,0 +1,18 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+export const getIndexPatternFromRawQuery = (query: string) : string => {
+  const indexMatches = query.match('\=(.*)');
```

Comment:
maybe use `/regex/` instead of `'regex'`, also the `\` and `()` doesn't seem to make a difference? 

does this work with the example i gave before? I was thinking something like `.match(/source\s*=\s*([^|\s]+)/)[1]` instead of js string manipulation

Reply:
Sure, we can use /regex/. Yes, the \ and () can be removed in this case. 

This works with the example you gave before and also other corner cases. But I agree it's better to use regex entirely in this case.

Reply:
Got it, then looks good. up to you on which implementation you want to use.

just a note i did not test the regex i wrote. if you use something similar maybe also add the `/i` flag to ignore case, and i think `index` can be used instead of `source` so maybe `(source|index)`

Reply:
Exactly, I'm adding index and /i to it, thanks